pkg/controllers/disruption/consolidation_test.go

```diff
@@ -3508,58 +3446,530 @@ var _ = Describe("Consolidation", func() {
 				ExpectReconcileSucceeded(ctx, disruptionController, client.ObjectKey{})
 			}()
 
-			// wait for the controller to block on the validation timeout
-			Eventually(fakeClock.HasWaiters, time.Second*5).Should(BeTrue())
-			// controller should be blocking during the timeout
-			Expect(finished.Load()).To(BeFalse())
+			// advance the clock so that the timeout expires for multi-nodeClaim
+			fakeClock.Step(disruption.MultiNodeConsolidationTimeoutDuration)
+			// advance the clock so that the timeout expires for single-nodeClaim
+			fakeClock.Step(disruption.SingleNodeConsolidationTimeoutDuration)
 
-			// and the node should not be deleted yet
-			ExpectExists(ctx, env.Client, nodeClaim1)
-			ExpectExists(ctx, env.Client, nodeClaim2)
-			ExpectExists(ctx, env.Client, nodeClaim3)
+			ExpectTriggerVerifyAction(&wg)
 
-			var extraPods []*v1.Pod
-			for i := 0; i < 2; i++ {
-				extraPods = append(extraPods, test.Pod(test.PodOptions{
-					ResourceRequirements: v1.ResourceRequirements{
-						Requests: v1.ResourceList{v1.ResourceCPU: *resource.NewQuantity(1, resource.DecimalSI)},
+			// controller should finish
+			Eventually(finished.Load, 10*time.Second).Should(BeTrue())
+			wg.Wait()
+
+			// should have no nodeClaims deleted from single nodeClaim consolidation
+			Expect(ExpectNodeClaims(ctx, env.Client)).To(HaveLen(numNodes))
+		})
+	})
+	Context("Multi-NodeClaim", func() {
+		var nodeClaimArray, spotNodeClaimArray []*v1beta1.NodeClaim
+		var nodeArray, spotNodeArray []*v1.Node
+
+		BeforeEach(func() {
+			nodeClaimArray = []*v1beta1.NodeClaim{}
+			spotNodeClaimArray = []*v1beta1.NodeClaim{}
+			nodeArray = []*v1.Node{}
+			spotNodeArray = []*v1.Node{}
+			for i := 1; i <= 3; i++ {
+				tempNodeClaim, tempNode := test.NodeClaimAndNode(v1beta1.NodeClaim{
```

Comment:
There's a test.NodeClaimsAndNodes() function that will do this for you

Reply:
Thanks. Re-used this.